erpnext/non_profit/doctype/member/member.js

```diff
@@ -21,12 +21,15 @@ frappe.ui.form.on('Member', {
 
 			// custom buttons
 			frm.add_custom_button(__('Accounting Ledger'), function() {
-				frappe.set_route('query-report', 'General Ledger',
-					{party_type:'Member', party:frm.doc.name});
+				if (frm.doc.customer) {
+					frappe.set_route('query-report', 'General Ledger', {party_type: 'Customer', party: frm.doc.customer});
+				} else {
+					frappe.set_route('query-report', 'Accounts Receivable', {party_type: 'Member', party: frm.doc.name});
+				}
 			});
 
 			frm.add_custom_button(__('Accounts Receivable'), function() {
-				frappe.set_route('query-report', 'Accounts Receivable', {member:frm.doc.name});
+				frappe.set_route('query-report', 'Accounts Receivable', {customer: frm.doc.customer});
 			});
```

Comment:
Can we hide this button if there's no customer set? Going to AR report without customer is pointless. Lets not give users the option unless valid

Reply:
Hi @marination,
As per your according, added a condition like if have a customer in member doctype then AR will appear.


https://user-images.githubusercontent.com/99652762/180206359-4b1ce1cf-97ee-42eb-a20d-86af9cdee385.mp4